plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/CloudLibraries.java

```diff
@@ -83,6 +84,29 @@ public static Library getLibrary(String id) {
         logger.log(Level.SEVERE, "Error loading library definition", ex);
       }
     }
-    return builder.build();
+
+    ImmutableMap<String, Library> map = builder.build();
+    
+    resolveTransitiveDependencies(map);
+    
+    return map;
+  }
+
+  // Only goes one level deeper, which is all we need for now.
+  // Does not recurse.
+  private static void resolveTransitiveDependencies(ImmutableMap<String, Library> map) {
+    for (Library library : map.values()) {
+      List<String> directDependencies = library.getLibraryDependencies();
+      List<String> transitiveDependencies = Lists.newArrayList(directDependencies);
```

Comment:
I avoid using these kinds of collection builders in Guava since they are not recommended anymore: https://google.github.io/guava/releases/21.0/api/docs/com/google/common/collect/Lists.html#newArrayList--